### PR TITLE
dma: Better suspend/resume testing

### LINKED ISSUE #44006: intel_adsp_cavs25: tests/drivers/dma/loop_transfer failed
**Describe the bug**
tests/drivers/dma/loop_transfer failed due to a assertion failure. (after this commit 652e374161d2e7c918f443012fecf4faae947b8f)

**To Reproduce**
1. twister -W -p intel_adsp_cavs25 --device-testing --device-serial-pty=$HOME/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstwist.sh --west-flash=$HOME/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstwist.sh -T tests/drivers/dma/loop_transfer/ -vvv

**Logs and console output**
```
INFO:cavs-fw:Mapped PCI bar 0 of length 16384 bytes.
INFO:cavs-fw:Selected output stream 7 (GCAP = 0x1009701)
INFO:cavs-fw:Mapped PCI bar 4 of length 1048576 bytes.
INFO:cavs-fw:Detected cAVS 1.8+ hardware
INFO:cavs-fw:Trimming 16 bytes of extended manifest
INFO:cavs-fw:Resetting HDA device
INFO:cavs-fw:Powering down DSP cores
INFO:cavs-fw:Configuring HDA stream 7 to transfer firmware image
INFO:cavs-fw:Mapped 2M huge page at 0x14cc00000 to contain 90112 bytes of firmware
INFO:cavs-fw:Filled the buffer descriptor list (BDL) for DMA.
INFO:cavs-fw:Starting DSP, ADSPCS = 0xf0f
INFO:cavs-fw:Wait for ROM startup
INFO:cavs-fw:Sending IPC command, HIPIDR = 0x81004000
INFO:cavs-fw:Starting DMA, FW_STATUS = 0x5000001
INFO:cavs-fw:Waiting for firmware handoff, FW_STATUS = 0x5000001
INFO:cavs-fw:cAVS firmware load complete
--
I: Device DMA_1 initialized
I: Device DMA_0 initialized
Running test suite dma_m2m_loop_test
===================================================================
START - test_dma_m2m_loop
DMA memory to memory transfer started on DMA_0
Preparing DMA Controller
this platform do not support the dma channel
Starting the transfer on channel 0 and waiting for 1 second
Each RX buffer should contain the full TX buffer string.
RX data Loop 0: The quick brown fox jumps over the lazy dog
RX data Loop 1: The quick brown fox jumps over the lazy dog
RX data Loop 2: The quick brown fox jumps over the lazy dog
RX data Loop 3: The quick brown fox jumps over the lazy dog
RX data Loop 4: The quick brown fox jumps over the lazy dog
Finished: DMA
PASS - test_dma_m2m_loop in 1.2 seconds
===================================================================
START - test_dma_m2m_loop_suspend_resume
DMA memory to memory transfer started on DMA_0
Preparing DMA Controller
this platform do not support the dma channel
Starting the transfer on channel 0 and waiting for 1 second
dma suspend not supported
Sleeping while suspended
ERROR: failed to suspend transfers
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/dma/loop_transfer/src/test_dma_loop.c:310: test_dma_m2m_loop_suspend_resume: (test_loop_suspend_resume() == TC_PASS) is false
FAIL - test_dma_m2m_loop_suspend_resume in 1.2 seconds
===================================================================
Test suite dma_m2m_loop_test failed.
===================================================================
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain zephyr-sdk-0.13.2
 - Commit 652e374161d2e7c918f443012fecf4faae947b8f

### LINKED ISSUE #44106: test of dma drivers fails on dma_m2m_loop_test
**Describe the bug**
the testcase : tests/drivers/dma/loop_transfer fails on all stm32 target boards on the test_loop_suspend_resume()

After the merge of the https://github.com/zephyrproject-rtos/zephyr/pull/43738/ the testcase fails because the stm32 dma has no suspend/resume mode, 
Probably also failing on any target where the dma_api_suspend suspend; 	dma_api_resume resume; are not implemented ( -ENOSYS If not implemented. returned) as described in the include/drivers/dma.h 

Please also mention any information which could help others to understand
the problem you're facing:
 - all the stm32 targets that can execute the tests/drivers/dma/loop_transfer test.
 - this comes as a regression after the change in the test_loop_suspend_resume() function of the tests/drivers/dma/loop_transfer/src/test_dma_loop.c

**To Reproduce**
1. west build -p auto -b nucleo_wb55rg tests/drivers/dma/loop_transfer/
3. west flash

**Expected behavior**
```
START - test_dma_m2m_loop_suspend_resume
DMA memory to memory transfer started on DMAMUX_1
Preparing DMA Controller
this platform do not support the dma channel
Starting the transfer on channel 11 and waiting for 1 second
Suspended transfers
dma suspend not supported
 PASS - test_dma_m2m_loop_suspend_resume in 0.21 seconds
===================================================================
Test suite dma_m2m_loop_test succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```
**Impact**
test campaign failure

**Logs and console output**
```
*** Booting Zephyr OS build zephyr-v3.0.0-1639-ga4d438444771  ***
Running test suite dma_m2m_loop_test
===================================================================
START - test_dma_m2m_loop
DMA memory to memory transfer started on DMAMUX_1
Preparing DMA Controller
this platform do not support the dma channel
Starting the transfer on channel 11 and waiting for 1 second
Each RX buffer should contain the full TX buffer string.
RX data Loop 0: The quick brown fox jumps over the lazy dog
RX data Loop 1: The quick brown fox jumps over the lazy dog
RX data Loop 2: The quick brown fox jumps over the lazy dog
RX data Loop 3: The quick brown fox jumps over the lazy dog
RX data Loop 4: The quick brown fox jumps over the lazy dog
Finished: DMA
 PASS - test_dma_m2m_loop in 1.49 seconds
===================================================================
START - test_dma_m2m_loop_suspend_resume
DMA memory to memory transfer started on DMAMUX_1
Preparing DMA Controller
this platform do not support the dma channel
Starting the transfer on channel 11 and waiting for 1 second
dma suspend not supported
E: Transfer Error.
I: tc: 0, ht: 0, te: 0, gi: 0

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/dma/loop_transfer/src/)
DMA could not proceed, an error occurred

 FAIL - test_dma_m2m_loop_suspend_resume in 0.40 seconds
===================================================================
Test suite dma_m2m_loop_test failed.
===================================================================
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - Commit SHA : 1d3dbd49e14e9dec14a6bfa4d69fa56b62406784

**Additional context**

### PR DESCRIPTION
**Pull request by teburd**
Make the test data larger to help slow down the transfer long enough
to get suspend to work. Then attempts to spin on suspend. If we can
suspend any of the test transfers, we succeed.

Previous suspend resume test would always fail on DesignWare DMA
as it was seemingly already done by the time dma_suspend would be
called.

Fixes #44006
Fixes #44106 

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

### CODE REVIEW

**Review by hakehuang — DISMISSED** [review]

File: tests/drivers/dma/loop_transfer/src/test_dma_loop.c [hunk 1]
```diff
@@ -247,20 +259,39 @@ static int test_loop_suspend_resume(void)
...
+			TC_PRINT("ERROR: transfer stop\n");
+		}
+		return TC_FAIL;
 	}
+	TC_PRINT("suspended after %d transfers occurred\n", transfer_count);
 
-	TC_PRINT("Sleeping while suspended\n");
-	k_sleep(K_MSEC(SLEEPTIME));
+	/* Now sleep */
+	k_msleep(100);
```
[the remark this answers is not in the document]
Comment (teburd) [reply tests/drivers/dma/loop_transfer/src/test_dma_loop.c:284]: Probably better with the define, unintentional, good eye.

**Review by hakehuang — DISMISSED** [review]

**Comment by LixinGuoX** [comment]
Verified on intel_adsp_cavs25.

**Review by ghost — DISMISSED** [review]

**Comment by teburd** [comment]
@FRASTM saw an issue on STM32 that should be resolved with the latest push

**Comment by laurenmurphyx64** [comment]
I can confirm it passes on the `disco_l475_iot1` too :+1:

**Review by laurenmurphyx64 — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]
that works for me too, tested on the nucleo_wb55rg or nucleo_h743zi (with DMAMUX) aand nucleo_l476rg (no DMAMUX)

**Review by ghost — APPROVED** [review]

**Pull request merged**